src/ocl/utilocl.cpp

```diff
@@ -699,107 +699,119 @@ float transpose_NCHW2CNHW(const Handle& handle,
+        c /= 4;
+        in_offset /= 4;
+        out_offset /= 4;
     }
-    else
+
+    if(h_stride == 1 && w_stride == 1 && type == miopenFloat)
     {
-        std::string params = GetDataTypeKernelParams(type);
+        kernel_name += "_V1";
 
-        if(type == miopenInt8x4)
+        int RD_BLCK      = ((h_in * w_in) % 4 == 0) ? 4 : ((h_in * w_in) % 2 == 0) ? 2 : 1;
+        int HW_RD        = (h_in * w_in) / RD_BLCK;
+        size_t MAP_RD    = HW_RD * c;
+        size_t lcl_size0 = WG_SIZE; //((MAP_RD + 63)/64 < 4) ? ((MAP_RD + 63)/64)*64 : 256;
+
+        std::string READ_TYPE = (RD_BLCK == 1) ? "float" : "float" + std::to_string(RD_BLCK);
+
+        const std::vector<size_t> vld{lcl_size0, 1, 1};
+        std::vector<size_t> vgd{MAP_RD, 1, 1};
+
+        if(MAP_RD < MAX_ACTIVE_THREADS)
         {
-            c /= 4;
-            in_offset /= 4;
-            out_offset /= 4;
+            vgd = {MAP_RD, static_cast<size_t>(n), 1};
+            kernel_name += "_2D_WG";
         }
-
-        if(h_stride == 1 && w_stride == 1 && type == miopenFloat)
+        else
         {
-            params +=
-                " -DNC_TRANS_NCHW_OPT=1 -DNC_TRANS_CNHW_OPT=0 -DNC_TRANS_NCHW=0 -DNC_TRANS_CNHW=0";
-
-            int RD_BLCK      = ((h_in * w_in) % 4 == 0) ? 4 : ((h_in * w_in) % 2 == 0) ? 2 : 1;
-            int HW_RD        = (h_in * w_in) / RD_BLCK;
-            size_t MAP_RD    = HW_RD * c;
-            size_t lcl_size0 = WG_SIZE; //((MAP_RD + 63)/64 < 4) ? ((MAP_RD + 63)/64)*64 : 256;
-
-            std::string READ_TYPE = (RD_BLCK == 1) ? "float" : "float" + std::to_string(RD_BLCK);
-
-            params += " -DIN_OFF=" + std::to_string(in_offset);
-            params += " -DOUT_OFF=" + std::to_string(out_offset);
-            params += " -DH=" + std::to_string(h_in);
-            params += " -DW=" + std::to_string(w_in);
-            params += " -DN=" + std::to_string(n);
-            params += " -DC=" + std::to_string(c);
-            params += " -DRD_BLCK=" + std::to_string(RD_BLCK);
-            params += " -DHW_RD=" + std::to_string(HW_RD);
-            params += " -DMAP_RD=" + std::to_string(MAP_RD);
-            params += " -DREAD_TYPE=" + READ_TYPE;
-
-            const std::vector<size_t> vld{lcl_size0, 1, 1};
-            std::vector<size_t> vgd{MAP_RD, 1, 1};
-
-            if(MAP_RD < MAX_ACTIVE_THREADS)
-            {
-                vgd = {MAP_RD, static_cast<size_t>(n), 1};
-                params += " -DIS_2D_WG=1";
-            }
-            else
-            {
-                params += " -DIS_2D_WG=0";
-            }
+            vgd = {MAP_RD, 1, 1};
+            kernel_name += "_1D_WG";
+        }
 
+        kernel_name += "_" + READ_TYPE;
+
+        auto&& kernels = handle.GetKernels(kernel_name, network_config);
+        if(!kernels.empty())
+        {
+            auto kernel = kernels.front();
+            kernel.ldims = {vld[0], vld[1], vld[2]};
+            kernel.gdims = {vgd[0], vgd[1], vgd[2]};
```

Comment:
@zjing14 Here is the fix to the issue you were seeing.

Reply:
I see. Thanks.